drivers: clock_control: stm32h7: Fix frequency calculation overflow

biglben opened the pull request:
STM32h7 pllout frequency calculation overflows. In the
worst case pllsrc_freq can be 50Mhz and plln_mul 512 which will cause
an overflow of the intermediate result which leads to wrong frequency
returned. As no intermediate result can be bigger than 960MHz only the
order of operations is changed.

Signed-off-by: Benjamin Bigler <benjamin.bigler@securiton.ch>

biglben committed:
drivers: clock_control: stm32h7: Fix frequency calculation overflow

STM32h7 pllout frequency calculation overflows. In the
worst case pllsrc_freq can be 50Mhz and plln_mul 512 which will cause
an overflow of the intermediate result which leads to wrong frequency
returned. As no intermediate result can be bigger than 960MHz only the
order of operations is changed.

Signed-off-by: Benjamin Bigler <benjamin.bigler@securiton.ch>

erwango reviewed (dismissed):
I think we have the same on U5 then. Would you mind fixing it as well ?
Approved anyway.

biglben committed:
drivers: clock_control: stm32u5: Fix frequency calculation overflow

STM32u5 pllout frequency calculation overflows. Fix by changing
order of operations

Signed-off-by: Benjamin Bigler <benjamin.bigler@securiton.ch>

biglben commented:
> 
Yes I can fix that as well
